pkg/controller/deployment/util/deployment_util.go

```diff
@@ -51,6 +51,10 @@ const (
 	// is deployment.spec.replicas + maxSurge. Used by the underlying replica sets to estimate their
 	// proportions in case the deployment has surge replicas.
 	MaxReplicasAnnotation = "deployment.kubernetes.io/max-replicas"
+	// FailedProgressAnnotation on a replica set denotes that it has failed to progress during its deployment.
+	FailedProgressAnnotation = "deployment.kubernetes.io/failed-progressing"
+	// CompleteProgressAnnotation on a replica set denotes that it has completed successfully its deployment.
+	CompleteProgressAnnotation = "deployment.kubernetes.io/complete-progressing"
```

Comment:
I want both of these annotations to be consistent. I am not sure about the names though. Since I am saving a timestamp as a value how about `failed-at` + `completed+at`? This feels more like Conditions on the replica set tbh..